src/main/java/org/snomed/snowstorm/mrcm/MRCMUpdateService.java

```diff
@@ -66,32 +66,49 @@ public class MRCMUpdateService extends ComponentService implements CommitListene
 
 	@Override
 	public void preCommitCompletion(Commit commit) {
-		if (BranchMetadataHelper.isImportingCodeSystemVersion(commit)) {
-			logger.info("MRCM auto update is disabled on branch {}", commit.getBranch().getPath());
+		Branch branch = commit.getBranch();
+		String branchPath = branch.getPath();
+
+		if (BranchMetadataHelper.isImportingCodeSystemVersion(commit) || BranchMetadataHelper.isSkipMrcmUpdateServiceCommit(commit)) {
+			logger.info("MRCM auto update is disabled on branch {}", branchPath);
 			return;
 		}
+
 		if (commit.getCommitType() == CONTENT) {
-			logger.debug("Start updating MRCM domain templates and attribute rules on branch {}.", commit.getBranch().getPath());
+			logger.debug("MRCM auto update starting for branch {}.", branchPath);
 			try {
-				performUpdate(false, commit);
-				logger.debug("End updating MRCM domain templates and attribute rules on branch {}.", commit.getBranch().getPath());
+				performUpdate(false, true, isUpdateMRCMDomainTemplates(branch), commit);
+				logger.debug("MRCM auto update ending for branch {}.", branchPath);
 			} catch (Exception e) {
-				throw new IllegalStateException("Failed to update MRCM domain templates and attribute rules." + e, e);
+				throw new IllegalStateException("Failed to run MRCM auto update for branch " + branchPath, e);
 			}
 		}
 	}
 
 	public void updateAllDomainTemplatesAndAttributeRules(String path) throws ServiceException {
 		logger.info("Updating all MRCM domain templates and attribute rules on branch {}.", path);
 		try (Commit commit = branchService.openCommit(path, branchMetadataHelper.getBranchLockMetadata("Updating all MRCM components."))) {
-			performUpdate(true, commit);
+			BranchMetadataHelper.skipMrcmUpdateServiceCommit(commit); // Transient property to skip own preCommitCompletion.
+			performUpdate(true, true, true, commit);
```

Comment:
I noticed this admin method would call `performUpdate` twice. Once directly, and then once indirectly once the `preCommitCompletion` method triggered. This transient flag prevents that.